sql/src/main/java/io/crate/expression/scalar/arithmetic/ArithmeticFunctions.java

```diff
@@ -39,19 +43,31 @@
 import io.crate.types.DoubleType;
 import io.crate.types.FloatType;
 import io.crate.types.IntegerType;
+import io.crate.types.IntervalType;
 import io.crate.types.LongType;
 import io.crate.types.ShortType;
 import io.crate.types.TimestampType;
+import org.elasticsearch.common.util.set.Sets;
+import org.joda.time.DateTime;
+import org.joda.time.DateTimeZone;
+import org.joda.time.Duration;
+import org.joda.time.Period;
 
+import javax.annotation.Nullable;
+import java.math.BigDecimal;
+import java.math.MathContext;
+import java.math.RoundingMode;
 import java.util.Arrays;
 import java.util.List;
+import java.util.Locale;
 import java.util.Set;
+import java.util.function.BiFunction;
 import java.util.function.BinaryOperator;
 
 public class ArithmeticFunctions {
 
     private static final Param ARITHMETIC_TYPE = Param.of(
-        DataTypes.NUMERIC_PRIMITIVE_TYPES, DataTypes.TIMESTAMPZ, DataTypes.TIMESTAMP);
+        DataTypes.NUMERIC_PRIMITIVE_TYPES, DataTypes.TIMESTAMPZ, DataTypes.TIMESTAMP, DataTypes.INTERVAL, DataTypes.UNDEFINED);
```

Comment:
Why do we need `UNDEFINED` here?

Reply:
To handle values in the arithmetics e.g.  `null + interval '1 second'`.

Reply:
This shouldn't be necessary, something like `null::timestamp + interval '1 second'` should work without adding undefined here.

Reply:
Ok, according to psql `null + interval `1 second` is supposed be null, but right now only works if I add `UNDEFINED` here. Is there a better way to handle this case ?

Reply:
What is the error message in `null + interval '1 second'` if you remove undefined here?

null/undefined can be casted to anything, so it should resolve to a signature that is more meaningful, e.g. `[timestamp, interval]`